### DISCUSSION TITLE
Request for Enhanced Documentation on Cart Transform API and Bundles

### DISCUSSION THREAD

**Discussion by Nem791** [Shopify/function-examples#268] [Feedback: Cart Transform API]
Hello Shopify Team,
I'm currently researching and try to implement Shopify Bundles as well as Cart Transform. I have encountered difficulties in fully understanding and utilizing these features due to the limitations of the current documentation. Here are some examples:

- In this step, the docs don't state where to put the Resolve input in and where to get the response: https://shopify.dev/docs/apps/selling-strategies/bundles/add-a-customized-bundle#step-1-define-the-input-of-your-function
- When I generate the cart_transform extension, it runs just fine without this mutation: https://shopify.dev/docs/apps/selling-strategies/bundles/ui#step-1-generate-a-new-extension
- I also don't understand why Bundle UI extension has to be requested instead of just showing it to everyone: https://shopify.dev/docs/apps/selling-strategies/bundles/ui#step-1-generate-a-new-extension

As far as I know, there is a cart_transform extension that helps merge and expand variants through metafields. I have generated and have some success in merging and expanding. But it would be nice if the docs explain explicitly how to apply graphQL API to modify metafields for bundles, or at least list them out.

My intention is not to criticize the current documentation, but rather to advocate for enhanced clarity and comprehensive explanations. Forgive me if I unintentionally overlooked any crucial details in my request.

**Comment by BrandoCommando** [comment]
> * In this step, the docs don't state where to put the Resolve input in and where to get the response

I would first go through the [Discount function tutorials](https://shopify.dev/docs/apps/selling-strategies/discounts) to get a handle on QueryInput and FunctionResult.

> * I also don't understand why Bundle UI extension has to be requested instead of just showing it to everyone: https://shopify.dev/docs/apps/selling-strategies/bundles/ui#step-1-generate-a-new-extension

I think the UI extension is "not quite ready". It takes a while to get access (at least for me), and then [doesn't work](https://community.shopify.com/c/extensions/cant-use-admin-product-configuration-render-extension-point/td-p/2103319). Avoid this step altogether.

> But it would be nice if the docs explain explicitly how to apply graphQL API to modify metafields for bundles, or at least list them out

Are you talking about returning product/variant metafields, or a configuration metafield? For CartTransform configuration, you have to use a MetafieldsSet mutation (with ownerId set to "gid://shopify/CartTransform/######"). [Discount function creation mutations](https://shopify.dev/docs/api/admin-graphql/unstable/mutations/discountCodeAppCreate) allow metafields to be set upon creation, but CartTransformCreate does not yet allow that.

> My intention is not to criticize the current documentation, but rather to advocate for enhanced clarity and comprehensive explanations. Forgive me if I unintentionally overlooked any crucial details in my request.

I think the Bundles functionality is still "under development", and typically fully fledged documentation comes after functionality becomes _stable_. I think the Shopify devs are looking more for use-case scenarios that will help them figure out a path to improve functionality for different uses.
